clients/python-wrapper/lakefs/object.py

```diff
@@ -357,62 +307,96 @@ def _get_range_string(start, read_bytes=None):
+    implicitly when using writer as a context.
     """
+    _fd: tempfile.SpooledTemporaryFile
+    _obj_stats: ObjectStats = None
+
+    def __init__(self,
+                 obj: StoredObject,
+                 mode: WriteModes,
+                 pre_sign: Optional[bool] = None,
+                 content_type: Optional[str] = None,
+                 metadata: Optional[dict[str, str]] = None,
+                 client: Optional[Client] = DEFAULT_CLIENT) -> None:
 
-    def __init__(self, repository: str, reference: str, path: str, client: Optional[Client] = DEFAULT_CLIENT) -> None:
-        super().__init__(repository, reference, path, client=client)
+        if 'x' in mode and obj.exists():  # Requires explicit create
+            raise ObjectExistsException
 
-    def upload(self,
-               data: str | bytes,
-               mode: WriteModes = 'wb',
-               pre_sign: Optional[bool] = None,
-               content_type: Optional[str] = None,
-               metadata: Optional[dict[str, str]] = None) -> WriteableObject:
-        """
-        Creates a new object or overwrites an existing object
-        :param data: The contents of the object to write (can be bytes or string)
-        :param mode: Write mode:
-            'x'     - Open for exclusive creation
-            'xb'    - Open for exclusive creation in binary mode
-            'w'     - Create a new object or truncate if exists
-            'wb'    - Create or truncate in binary mode
-        :param pre_sign: (Optional) Explicitly state whether to use pre_sign mode when uploading the object.
-        If None, will be taken from pre_sign property.
-        :param content_type: (Optional) Explicitly set the object Content-Type
-        :param metadata: (Optional) User metadata
-        :return: The Stat object representing the newly created object
-        :raises
-            ObjectExistsException if object exists and mode is exclusive ('x')
-            ObjectNotFoundException if repo id, reference id or object path does not exist
-            PermissionException if user is not authorized to perform this operation, or operation is forbidden
-            ServerException for any other errors
-        """
         if mode not in get_args(WriteModes):
             raise ValueError(f"invalid write mode: '{mode}'. WriteModes: {WriteModes}")
 
-        if 'x' in mode and self.exists():  # Requires explicit create
-            raise ObjectExistsException
+        self.content_type = content_type
+        self.metadata = metadata
 
-        binary_mode = 'b' in mode
-        if binary_mode and isinstance(data, str):
-            content = data.encode('utf-8')
-        elif not binary_mode and isinstance(data, bytes):
-            content = data.decode('utf-8')
+        open_kwargs = {
+            "encoding": "utf-8" if 'b' not in mode else None,
+            "mode": 'wb+' if 'b' in mode else 'w+',
+            "buffering": _BUFFER_SIZE,
```

Comment:
Why mess with how python writes to the temp file once we go over the threshold for opening one? 

I believe the default behavior is to align with storage block size